services/component-registry/developer-guide.mdx

```diff
@@ -0,0 +1,204 @@
+  Email: your@email.com
+```
+
+And if you need assistance about a CLI command:
+
+```bash 
+$ edgee help
+Usage: edgee [OPTIONS] <COMMAND>
+Commands:
+  login   Log in to the Edgee Console
+  whoami  Print currently login informations
+  serve   Run the Edgee server [aliases: server]
+  test    TBD
+  check   TBD
+```
+
+#### Create a new component
+
+Next, you can create a new component project and choose your preferred programming language.
+This will create a new local folder with the sample code and tooling you need to get started quickly.
+
+```bash 
+$ edgee component create
+Choose a name: my-new-component
+Choose a language from the sample projects below: example-rust-component
+```
+
+Now you can enter the new folder and start implementing your business logic.
+
+In short, a data collection component needs to implement this WIT interface:
+
+```text
+page: func(e: event, settings: dict) -> result<edgee-request, string>;
+track: func(e: event, settings: dict) -> result<edgee-request, string>;
+user: func(e: event, settings:dict) -> result<edgee-request, string>;
+```
+
+The corresponding method is invoked when a new `page`, `track` or `user` event is sent for data collection
+by the Edgee proxy. These methods receive the incoming event object and a set of settings, and the expected
+output is an `edgee-request` object, which looks like this:
+
+```text
+record edgee-request {
+    method: http-method,
+    url: string,
+    headers: dict,
+    forward-client-headers: bool,
+    body: string,
+}
+```
+
+<Note>
+  Note: an Edgee component doesn't perform any HTTP call on its own because it runs in a sandboxed environment with no access
+  to network or file system. The inteded purpose of a component is to instruct the proxy on which HTTP request to perform
+  to fullfil its data collection duties.
+</Note>
+
+For example, the Google Analytics component returns an `edgee-request` object that points to the official
+`https://www.google-analytics.com/g/collect` endpoint, with the correct querystring parameters.
+Similarly, the Amazon S3 component points to `https://{bucket}.s3.amazonaws.com` with the correct sigv4 authentication headers.
+
+In order to make your Edgee component reusable by other developers and organizations via the Edgee Component Registry,
+you can define a set of input `settings`, containing all the API credentials or dynamic
+information you need to craft the correct `edgee-request` object. For example, the Google Analytics component only expects a
+`ga_measurement_id` setting, while the Segment component expects `segment_project_id` and `segment_write_key`.
+
+You can configure the expected settings for your component in the local manifest file (TODO add more details about this).
+
+#### Build and test locally
+
+When the implementation is ready, you can build your component locally using the CLI:
+
+```bash
+$ edgee component build
+```
+
+The output of this command will be a new .wasm file in the current folder.
```

Comment:
i dont think that's true, we dont copy it in the current folder :thinking:

Reply:
maybe we should? :)